include/verilated_funcs.h

```diff
@@ -2228,64 +2228,261 @@ static inline WDataOutP VL_SEL_WWII(int obits, int lbits, WDataOutP owp, WDataIn
     return owp;
 }
 
+template <typename T>
+static inline VlQueue<T> VL_CLONE_Q(const VlQueue<T>& from) {
+    VlQueue<T> ret;
+    ret.renew(from.size());
+    for (size_t i = 0; i < from.size(); ++i) ret.atWrite(i) = from.at(i);
+    return ret;
+}
+
+template <typename T>
+static inline VlQueue<T> VL_REVCLONE_Q(const VlQueue<T>& from, int lbits, int elementBits) {
+    VlQueue<T> ret;
+    VL_REVCOPY_Q(ret, from, lbits, elementBits);
+    return ret;
+}
+
+template <typename T>
+static inline void VL_COPY_Q(VlQueue<T>& q, const VlQueue<T>& from) {
+    q.renew(from.size());
+    for (size_t i = 0; i < from.size(); ++i) q.atWrite(i) = from.at(i);
+}
+
+// Helper function to extract a bit from a flat bit stream represented by a queue
+template <typename T>
+static inline bool VL_GET_QUEUE_BIT(const VlQueue<T>& queue, int elementBits,
+                                    size_t bitIndex) VL_PURE {
+    const size_t elementIndex = bitIndex / elementBits;
+    const size_t bitInElement = bitIndex % elementBits;
+    if (elementIndex >= queue.size()) return false;
```

Comment:
Likewise move `if` and add UNLIKELY in other cases.

```suggestion
    if (VL_UNLIKELY(elementIndex >= queue.size())) return false;
    const size_t elementIndex = bitIndex / elementBits;
    const size_t bitInElement = bitIndex % elementBits;
```